File: fraction.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    bool isCoprime (int a, int b) {
        int gcd = __gcd(a,b);
        return (gcd == 1);
    }
    vector<int> properFraction (int sum) {
        int numerator = sum / 2;
        int denominator = (sum + 1) / 2;
        vector<int> result;
        while (!isCoprime(numerator, denominator)) {
            numerator -= 1;
            denominator += 1;
        }
        result.push_back(numerator);
        result.push_back(denominator);
        return result;
    }
};

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);cout.tie(nullptr);
    #ifndef ONLINE_JUDGE
        freopen("input.txt","r",stdin);
        freopen("output.txt","w",stdout);
    #endif
    Solution ans;
    int given_sum;
    cin >> given_sum;
    vector<int> result = ans.properFraction(given_sum);
    cout<< result[0] << " " <<result[1] << endl;
    return 0;
}

File: Java/Fraction.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class Fraction {
    public static int gcd(int a, int b) {
        if (b == 0) {
            return a;
        }
        return gcd(b, a % b);
    }

    public static boolean isCoprime(int a, int b) {
        int gcd = gcd(a, b);
        return (gcd == 1);
    }

    public static List<Integer> properFraction(int sum) {
        int numerator = sum / 2;
        int denominator = (sum + 1) / 2;
        List<Integer> result = new ArrayList<>();
        while (!isCoprime(numerator, denominator)) {
            numerator -= 1;
            denominator += 1;
        }
        result.add(numerator);
        result.add(denominator);
        return result;
    }
    
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        int givenSum = scan.nextInt();
        List<Integer> result = properFraction(givenSum);
        System.out.println(result.get(0) + " " + result.get(1));
        scan.close();
    }
}

